File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/World.java
package io.github.alexiscomete.lapinousecond.worlds;

public abstract class World {
    int travelPrice;
    String name, nameRP, progName, desc;

    public World(int travelPrice, String name, String nameRP, String progName, String desc) {
        this.travelPrice = travelPrice;
        this.name = name;
        this.nameRP = nameRP;
        this.progName = progName;
        this.desc = desc;
    }

    public abstract double getPriceForDistance(double distance, boolean place);
    public abstract double getDistance(Place place1, Place place2);
}

File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/World.kt
package io.github.alexiscomete.lapinousecond.worlds

import io.github.alexiscomete.lapinousecond.worlds.map.Map
import java.awt.image.BufferedImage
import java.io.IOException
import javax.imageio.ImageIO

open class World(
    val typeOfServer: String,
    val name: String,
    val progName: String,
    val desc: String,
    val defaultX: Int,
    val defaultY: Int,
    val mapPath: String,
    val mapWidth: Int,
    val mapHeight: Int,
    val map: Map,
) {
    val mapFile: BufferedImage? = try {
        ImageIO.read(
            World::class.java.classLoader.getResourceAsStream(mapPath)
        )
    } catch (e: IOException) {
        e.printStackTrace()
        null
    }
}